### DISCUSSION TITLE
Easy way of running n Instances?

### DISCUSSION THREAD

**Discussion by Suven** [komoot/photon#692] [Q&A]
Hey,

I downloaded the search index, threw into into a persistent docker-volume an started photon in my k8s-cluster inside a pod. That works just like local for a single instance, although when running a second pod, it fails because the elastic-index is already locked by the first pod. 

Is there some easy way to use the downloaded index and horizontal scaling?

**Comment by lonvia** [comment]
There are two ways to go about it:

1. Simply duplicate your docker-volumes and run a separate, independent database on each. Unless you run updates, the database is completely static. So you get identical results from all instances. In this setup you can add, remove and update instances easily without affecting the rest of your running instances.

2. Set up an instance of ElasticSearch 5.6 (the version is important, we are unfortunately way behind the current version) and point it to the photon data dump. Then run photon in external database mode using the `-transport-addresses` parameter. You can now use all the scaling features of ES.

**Reply by Suven** [reply]
Thanks for your reply!

2. Do you have experiences what the bottle-neck regarding the search-api is? Is it likely that I need to scale the ES-DB, photon, or both proportionally? I found that reverse-lookups are always fast but searching relly does not like concurrent requests :'D

**Reply by lonvia** [reply]
I don't have much experience with that, I'm afraid. Maybe other people who run Photon in bigger setups could comment?
